### PR TITLE
test: Rewrite screenshot review tool

### PR DESCRIPTION
**Pull request by joeyparrish**
This rewritten screenshot review tool is more dynamic and better organized.

It is now easier to add additional test suites, and filtering can be done dynamically by any attribute (platform, suite, or test).

Related to work on PR #4767

### CODE REVIEW

**Comment by joeyparrish** [comment]
Here's a screenshot of the new review tool:

review tool screenshot

File: test/test/assets/screenshots/review.html [hunk 1]
```diff
@@ -128,127 +131,244 @@
...
-  document.body.addEventListener('keyup', keyHandler);
-  window.addEventListener('blur', focusHandler);
-  window.addEventListener('focus', focusHandler);
+  return container;
+}
 
-  const appendScreenshot = (labelText, platform, screenshot) => {
-    const container = document.createElement('div');
-    container.classList.add('container');
+function appendScreenshot(platform, suite, test) {
```
Comment (theodab) [inline test/test/assets/screenshots/review.html:251]: Hm. I'm not sure if I like that the `test` parameter for this method is not the same as the `test` variable in the calling context. Instead, it's equal to `prefix` + `-` + `test`.
I feel like either this should have a different name (`prefixedTest` maybe?), the test variable in the main loop should have a different name, or at least this should have some JSDoc explaining that this wants the test name with prefix included.

**Review by theodab — CHANGES_REQUESTED** [review]

File: test/test/assets/screenshots/review.html (hunk 1 above)
Comment (joeyparrish) [reply test/test/assets/screenshots/review.html:251]: Good point.  Fixed.

File: test/test/assets/screenshots/review.html [hunk 2]
```diff
@@ -128,127 +131,244 @@
...
-    document.body.classList.remove('shift');
-    document.body.classList.remove('ctrl');
+function appendOption(parentElement, key, value, onChange) {
+  const container = document.createElement('div');
+  container.className = 'optionContainer';
+  parentElement.appendChild(container);
+
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+  input.dataset.key = key;
```
[the remark this answers is not in the document]
Comment (joeyparrish) [reply test/test/assets/screenshots/review.html:227]: I think I had plans to use this to make filtering decisions based on the state of the inputs, but I ended up using the keys of `allOptions` to get this metadata instead.  Removed.

**Review by theodab — APPROVED** [review]

**Pull request merged**